### ISSUE TITLE
DockerFile build issue

### ISSUE THREAD

**Issue by CurtMiller66** [MicrosoftDocs/visualstudio-docs#8123]
[Enter feedback here]
The documentation states that you can specify as an example:
  <PropertyGroup Condition="'$(Configuration)'=='Debug'">
   <DockerfileFile>DockerFileDebug</DockerfileFile>
  </PropertyGroup>

This doesn't work.  While you can adjust the  docker-compose.yml to point at the proper file, when you start Visiual Studio, you get the warning to "Please make sure you have a Dockerfile called 'Docerfile' adjacent to the project file."
More importantly, you get a fatal build error CTC1023 stating the same issue.

What got me into this is the fact that the build command for the dockerfile has a -c Release hard coded.  This means if you want to build something other than release, you have to remember to edit the file to point to the correct build configuration. Not very useful when you build various configurations for different systems.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 84f968ae-2fa3-06af-06c0-74e122071ce5
* Version Independent ID: af37447f-bb94-abac-c4cc-cac649575d2c
* Content: [Visual Studio Container Tools build properties - Visual Studio (Windows)](https://docs.microsoft.com/en-us/visualstudio/containers/container-msbuild-properties?view=vs-2022)
* Content Source: [docs/containers/container-msbuild-properties.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/containers/container-msbuild-properties.md)
* Product: **visual-studio-windows**
* Technology: **vs-container-tools**
* GitHub Login: @ghogen
* Microsoft Alias: **ghogen**

**Comment by WilliamAntonRohm** [comment]
@CurtMiller66 -- Curt, thank you for your feedback.  Please consider these resources:
- [Visual Studio on Microsoft Q&A](https://docs.microsoft.com/answers/products/vs) 
- [Visual Studio MSDN Forum](https://social.msdn.microsoft.com/Forums/vstudio) 
- [Stack Overflow](https://stackoverflow.com/search?q=%22visual+studio%22)  
- [Reddit](https://www.reddit.com/r/VisualStudio)

@ghogen -- Gordon, please look into this issue.

**Comment by CurtMiller66** [comment]
As a side note, my work around is to specify an environment variable to tell the code which configuration file to use which was my original goal with multiple configurations.  Config files were copied to dll.app.config as specified in the .csproj file.

**Comment by ghogen** [comment]
@CurtMiller66 
Do you mean DockerfileFile ?  That is the MSBuild property that is supposed to be used to specify a different Dockerfile. There isn't a DockerFileDebug property.

**Comment by CurtMiller66** [comment]
Not quite.  I tried App.Debug.Config, App.Canada.Config and App.US.Config.  As part of the csproj build process, the appropriate file was copied to App.Processor.Config and the application started.  So far, so good.  Then I added Docker support.  The build would fail with a missing file because of the hard coded -c Release instead of -c Debug, -c Canada or -c US which the csproj file expected.  My work around puts all the config files in the docker container and then specifies an environment variable when instantiate the container in Azure.  Problem solved for me but the documentation should state that the only version of code that can be put in a docker container is Release.

**Comment by ravipal** [comment]
@CurtMiller66 Thanks for clarifying your scenario. We are aware of the configuration is being hardcoded in the scaffolded Dockerfile during build and publish stage. Glad you found a way to use different configuration. Here is another way (using arguments) https://github.com/microsoft/DockerTools/issues/287#issuecomment-784489373 to do this. We have a proposed work item to handle this scenario.
